### PR TITLE
Refactor Models and Add PL-CLI Supported Models.

### PR DESCRIPTION
**Pull request by samet-akcay**
# Description

- This PR removes `hparams` from class constructor and uses explicit arguments, which will be automatically parsed by jsonargparse in pytorch lightning cli. 
- This is a prep PR for the upcoming CLI. 

## Changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist

- [x] My code follows the [pre-commit style and check guidelines](https://openvinotoolkit.github.io/anomalib/guides/using_pre_commit.html#pre-commit-hooks) of this project.
- [x] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing tests pass locally with my changes

### CODE REVIEW

File: anomalib/models/padim/model.py [hunk 1]
```diff
@@ -60,14 +55,12 @@ def __init__(
         layers: List[str],
         input_size: Tuple[int, int],
         backbone: str = "resnet18",
-        apply_tiling: bool = False,
-        tile_size: Optional[Tuple[int, int]] = None,
-        tile_stride: Optional[int] = None,
     ):
         super().__init__()
+        self.tiler: Optional[Tiler] = None
```
Comment (ashwinvaidya17) [inline anomalib/models/padim/model.py:60]: I like this refactor!

**Review by ashwinvaidya17 — APPROVED** [review]
Thanks! Looks good to me.

File: anomalib/models/components/base/anomalib_module.py [hunk 2]
```diff
@@ -0,0 +1,190 @@
+"""Base Anomalib Module for Train/Test/Inference Tasks."""
```
Comment (djdameln) [inline anomalib/models/components/base/anomalib_module.py:1]: Should we remove `AnomalyModule`? Or are you planning to do this in a later PR?

File: anomalib/models/patchcore/model.py [hunk 3]
```diff
@@ -244,24 +237,17 @@ class PatchcoreLightning(AnomalyModule):
...
-        input_size (Tuple[int, int]): Input size for the model.
-        tile_size (Tuple[int, int]): Tile size
-        tile_stride (int): Stride for tiling
-        backbone (str, optional): Pre-trained model backbone. Defaults to "resnet18".
-        apply_tiling (bool, optional): Apply tiling. Defaults to False.
+        hparams (Union[DictConfig, ListConfig]): Model params
     """
 
     def __init__(self, hparams) -> None:
+        warnings.warn("PadimLightning is deprecated, use Padim via Anomalib CLI instead", DeprecationWarning)
```
Comment (djdameln) [inline anomalib/models/patchcore/model.py:244]: typo: `PatchCoreLightning`

**Review by djdameln — COMMENTED** [review]
I'm not sure I fully understand these changes. Why are the lightning modules deprecated. Even when using the CLI, the Lightning modules are still needed to define the training and testing functionality right? Or am I missing something?

**Comment by samet-akcay** [comment]
> I'm not sure I fully understand these changes. Why are the lightning modules deprecated. Even when using the CLI, the Lightning modules are still needed to define the training and testing functionality right? Or am I missing something?

There is gonna be another PL Module, which will be introduced in the next PR. It will be almost the same as the existing ones. The only difference will be the input parameters of the classes, ie. hparams vs explicit class args.

File: anomalib/models/components/base/anomalib_module.py (hunk 2 above)
Comment (samet-akcay) [reply anomalib/models/components/base/anomalib_module.py:1]: yes, when we deprecate the existing lightning modules we'll deprecate anomaly_module as well

**Comment by samet-akcay** [comment]
@djdameln to be more specific, how the new lightning module will look like
https://github.com/openvinotoolkit/anomalib/blob/feature/sa/pl-cli-add-cli-scripts/anomalib/models/padim/model.py#L341

**Review by djdameln — APPROVED** [review]

**Pull request merged**